add qualifier

HealYouDown opened the pull request:
My qualifier for the summer code jam 2020.

Passing all tests except two additional basic and one additional advanced.

HealYouDown committed:
add qualifier

IgNoRaNt23 commented on qualifier-jeremy.py:
```diff
@@ -0,0 +1,92 @@
+        return obj.__dict__.get(self.name)
+
+    def __set__(self, obj, value) -> None:
+        if not isinstance(value, self.field_type):
```
it's often better to not negate (see this sentence as a good example). Positive statements are better understandable
Having if-else-blocks would also improve the readibility here

IgNoRaNt23 commented on qualifier-jeremy.py:
```diff
@@ -0,0 +1,92 @@
+
+        obj.__dict__[self.name] = value
+
+        if self.name == "content":
```
this logic is way better placed as a property of the article class. Dealing with special cases in this generic descriptor should be avoided

IgNoRaNt23 commented on qualifier-jeremy.py:
```diff
@@ -0,0 +1,92 @@
+        content: str
+    ):
+        # Assign unique id
+        self.id = Article.id_counter
```
this is problematic if you want a class (for example NewspaperArticle) to inherit from this class. Better use a @classmethod

IgNoRaNt23 commented on qualifier-jeremy.py:
```diff
@@ -0,0 +1,92 @@
+
+        # Create a list with tuples (word, count, index)
+        # to sort later
+        words = [(word, content_words.count(word), index)
```
If i understand correctly the index is for keeping the original order. It's not really needed since dictionaries are ordered since 3.7. See https://softwaremaniacs.org/blog/2020/02/05/dicts-ordered/ and you get the right order automatically

HealYouDown committed:
fix for the small things noted in the code-review

- Removed the negation of the check in the ArticleField class
- Moved setting last_edited from ArticleField to a setter
- Removed unncessecary index from list comprehension in most_common_words
- Removed hardcoding of Article class name to increase the id_counter
